src/capture/cleaner_middleware/cleaner_middleware.h

```diff
@@ -26,47 +26,13 @@
 #ifndef CLEANER_MIDDLEWARE_H
 #define CLEANER_MIDDLEWARE_H
 
-#include <sqlite3.h>
-#include <pcap.h>
-#include <stdint.h>
-
-#include "../../utils/allocs.h"
-#include "../../utils/os.h"
-#include "../../utils/eloop.h"
-
-/**
- * @brief Initialises the cleaner middleware
- *
- * @param db The sqlite3 db
- * @param db_path The sqlite3 db path
- * @param eloop The eloop structure
- * @param pc The pcap context
- * @return struct middleware_context* the middleware context on success, NULL on
- * failure
- */
-struct middleware_context *init_cleaner_middleware(sqlite3 *db, char *db_path,
-                                                   struct eloop_data *eloop,
-                                                   struct pcap_context *pc);
+#include "../middleware.h"
 
 /**
- * @brief Cleaner processors
- *
- * @param context The middleware context
- * @param ltype The packet type
- * @param header The pcap packet header
- * @param packet The pcap packet
- * @param ifname The capture interface
- * @return int 0 on success, -1 on failure
+ * @brief Cleaner Middleware.
+ * @author Alexandru Mereacre, Alois Klink
+ * The cleaner middleware is designed to periodically remove the oldest
+ * PCAP files when the use more than `CLEANER_STORE_SIZE` KiB.
```

Comment:
I think this is what the cleaner middleware does, but you might want to double-check this description.